youtube_dl/extractor/heise.py

```diff
@@ -54,6 +55,9 @@ class HeiseIE(InfoExtractor):
     }]
 
     def _real_extract(self, url):
+        url = re.sub(r'https?://(?:www\.|m\.)?heise\.de/',
```

Comment:
Should match from the beginning. It's senseless to replace mobile URL with itself.